haskell/test/Test/Ligo/BaseDAO/Proposal/Delegate.hs

```diff
@@ -0,0 +1,69 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.Ligo.BaseDAO.Proposal.Delegate
+  ( test_UpdateDelegates
+  ) where
+
+import Universum
+
+import Lorentz hiding (assert, (>>))
+import Morley.Nettest
+import Morley.Nettest.Tasty (nettestScenarioOnEmulatorCaps)
+import Test.Tasty (TestTree, testGroup)
+import Util.Named
+
+import Ligo.BaseDAO.Types
+import Test.Ligo.BaseDAO.Common
+import Test.Ligo.BaseDAO.Proposal.Config
+
+test_UpdateDelegates :: TestTree
+test_UpdateDelegates =
+  testGroup "Update_delegates:"
+    [ nettestScenarioOnEmulatorCaps "add/remove a delgate" $
+        addRemoveDelegate (originateLigoDaoWithConfigDesc dynRecUnsafe)
+    ]
+
+addRemoveDelegate
+  :: (MonadNettest caps base m, HasCallStack)
+  => (ConfigDesc Config -> OriginateFn m)
+  -> m ()
+addRemoveDelegate originateFn = do
+  DaoOriginateData{..} <- originateFn testConfig defaultQuorumThreshold
+  let param enable = DelegateParam
+        { dpEnable = enable
+        , dpDelegate = dodOperator1
+        }
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Update_delegate") [param True]
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Freeze") (#amount .! 100)
+
+  advanceLevel dodPeriod
+  let params counter = ProposeParams
+        { ppFrozenToken = 10
+        , ppProposalMetadata = lPackValueRaw @Integer counter
+        , ppFrom = dodOwner1
+        }
+      key = makeProposalKey (params 1)
+      vote = NoPermit VoteParam
+                { vVoteType = True
+                , vVoteAmount = 1
+                , vProposalKey = key
+                , vFrom = dodOwner1
+                }
+  withSender dodOperator1 $ call dodDao (Call @"Propose") (params 1)
+  advanceLevel dodPeriod
+  withSender dodOperator1 $ call dodDao (Call @"Vote") [vote]
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Update_delegate") [param False]
+
+  withSender dodOperator1 $ call dodDao (Call @"Vote") [vote]
+    & expectCustomErrorNoArg #nOT_DELEGATE dodDao
+  advanceLevel dodPeriod
+  withSender dodOperator1 $ call dodDao (Call @"Propose") (params 2)
+    & expectCustomErrorNoArg #nOT_DELEGATE dodDao
+
```

Comment:
A couple more things that would be nice to check:
1. that one can update multiple delegates
2. that (since it is a list) one can add and remove a delegate in the same call without an error
3. that one can remove a delegate even if it was not set before (since we are permissive about that)

Reply:
Added a new test that covers all this cases.